lib/scope.js

```diff
@@ -2485,10 +2485,39 @@ module.exports = {
       if ( !elem ) { reports.addToReport(scope, { type: `error`, value: msg }); }
       expect( elem, msg ).to.exist;
 
-      scope.toDownload = filename;
-      // Should this be using `scope.tapElement()`?
-      await elem.evaluate( elem => { return elem.click(); });
-      await scope.detectDownloadComplete( scope );
+      let failed_to_download = false;
+      let err_msg = "";
+      try {
+        const binaryStr = await scope.page.evaluate(el => {
+          const url = el.getAttribute("href");
+          return new Promise(async (resolve, reject) => {
+            const response = await fetch(url, {method: "GET"});
+            const reader = new FileReader();
+            reader.readAsBinaryString(await response.blob());
+            reader.onload = () => resolve(reader.result);
+            reader.onerror = () => reject(`Error occurred on page when downloading ${ url }: ${ reader.error }`);
+          });
+        }, elem);
+        if (binaryStr !== '') {
+          const fileData = Buffer.from(binaryStr, 'binary');
+          fs.writeFileSync(`${ scope.paths.scenario }/${ filename }`, fileData);
+          reports.addToReport(scope, { type: `info`, value: `Downloaded ${ filename } (${ fileData.length } bytes) to ${ scope.paths.scenario }`});
+        } else {
+          failed_to_download = true;
+          err_msg = `Couldn't download ${ url }, dinary data for download was empty`;
```

Comment:
```suggestion
          err_msg = `Couldn't download ${ url }, binary data for download was empty`;
```

Reply:
Or

```suggestion
          err_msg = `Couldn't download ${ filename }, binary data for download was empty`;
```

(I might have missed it, but I don't think `url` exists in this scope)